cantools/database/can/c_source.py

```diff
@@ -403,6 +402,27 @@
 bool {database_name}_{message_name}_{signal_name}_is_in_range({type_name} value);
 '''
 
+MESSAGE_DECLARATION_INIT_FMT = '''\
+/**
+ * Init message fields to default values from {database_message_name}.
+ *
+ * @param[in] msg_p Message to init.
+ *
+ * @return zero(0) on success or (-1) in case of nullptr argument.
+ */
+int {database_name}_{message_name}_init(struct {database_name}_{message_name}_t *msg_p);
+'''
+
+MESSAGE_DEFINITION_INIT_FMT = '''\
+int {database_name}_{message_name}_init(struct {database_name}_{message_name}_t *msg_p)
+{{
+    if (msg_p == NULL) return -1;
+    memset(msg_p, 0, sizeof(msg_p));
```

Comment:
this is incorrect: `sizeof(msg_p)` is the size of the pointer itself, not the object it points to:

```suggestion
    memset(msg_p, 0, sizeof(*msg_p));
```

Reply:
Hello.

Thank for pointing it to me.
It's not clear for me, how to deal with tests related to comparing generated files to files from tests/files_c_source directory.
Should I manually update files in tests/files_c_source ?

Reply:
> Should I regenerate source files using dbc's from tests/files/dbc and then place the into tests/c_source/ ?

basically yes. Maybe the tests should test the API of the generated source files instead of comparing them  byte-for-byte, but that's a topic for another day...

Reply:
Thank you.
So, I'll regenerate source files from .dbc's and see how it will work

Reply:
Hello.
I'm still afraid of replacing existing files from tests/c_source/.
How do you think, maybe it will be better to add additional flag like `--gen-initial-values` to cantools?
In this case initializing functions will be added to C sources only in case of this flag is passed to the tool.
All the old tests will still work properly.
I'll have to generate new files with initializing functions inside and write new tests for those files.
But these new files and tests will not replace existing ones.

Reply:
> I'm still afraid of replacing existing files from tests/c_source/

this is pretty easy to do: remove the comment from https://github.com/cantools/cantools/blob/master/tests/test_command_line.py#L73 and run test_command_line:
```bash
pytest tests/test_command_line.py
```

this regenerates all C source files. if everything looks okay after inspecting what's changed via git , add the comment back to the test and commit the modified files.

> How do you think, maybe it will be better to add additional flag like --gen-initial-values to cantools?

I'd prefer not to add too many knobs like this, in particular because I don't think that there are any notable disadvantages in having these functions around even if they are unused ...

Reply:
Okay, thank you.
I'll try this approach